cmake/CMakeLists.txt

```diff
@@ -2129,25 +1333,29 @@ else()
   list(APPEND onnxruntime_EXTERNAL_LIBRARIES ${CMAKE_DL_LIBS} Threads::Threads)
 endif()
 
-# When GDK_PLATFORM is set then WINAPI_FAMILY is defined in gdk_toolchain.cmake (along with other relevant flags/definitions).
-if (WIN32 AND NOT GDK_PLATFORM)
-  add_compile_definitions(WINAPI_FAMILY=100) # Desktop app
-  if (onnxruntime_USE_WINML)
-    add_compile_definitions(WINVER=0x0602 _WIN32_WINNT=0x0602 NTDDI_VERSION=0x06020000)  # Support Windows 8 and newer
+if (CMAKE_SYSTEM_NAME STREQUAL "Android")
+  list(APPEND onnxruntime_EXTERNAL_LIBRARIES log)
+endif()
+
+# check if we need to link against librt on Linux
+include(CheckLibraryExists)
+include(CheckFunctionExists)
+if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+  check_library_exists(rt clock_gettime "time.h" HAVE_CLOCK_GETTIME)
+
+  if (NOT HAVE_CLOCK_GETTIME)
+    set(CMAKE_EXTRA_INCLUDE_FILES time.h)
+    check_function_exists(clock_gettime HAVE_CLOCK_GETTIME)
+    set(CMAKE_EXTRA_INCLUDE_FILES)
   else()
-    add_compile_definitions(WINVER=0x0601 _WIN32_WINNT=0x0601 NTDDI_VERSION=0x06010000)  # Support Windows 7 and newer
-  endif()
-  if (NOT CMAKE_CXX_STANDARD_LIBRARIES MATCHES kernel32.lib)
-    # On onecore, link to the onecore build of the MSVC runtime
-    get_filename_component(msvc_path "${CMAKE_C_COMPILER}/../../../.." ABSOLUTE)
-    link_directories(BEFORE "${msvc_path}/lib/onecore/${onnxruntime_target_platform}")
-    # The .lib files in the MSVC runtime have a DEFAULITLIB entry for onecore.lib, which in turn links to reverse forwarders.
-    # We ignore that entry and use onecore_apiset.lib instead, since system components must not rely on reverse forwarders.
-    add_link_options("/NODEFAULTLIB:onecore.lib")
+    list(APPEND onnxruntime_EXTERNAL_LIBRARIES rt)
   endif()
 endif()
 
-include(wil.cmake)
+
+#Now the 'onnxruntime_EXTERNAL_LIBRARIES' variable should be sealed. It will be used in onnxruntime.cmake which will be included in the next.
+#The order of the following targets matters. Right depends on left. If target A appears before target B. Then A.cmake can not use variables defined in B.cmake.
```

Comment:
do we need to keep this ordering synchronized with onnxruntime.cmake's onnxruntime_INTERNAL_LIBRARIES?

Reply:
No. This is regarding to the dependency relationship between the cmake files. For example, if a cmake var is defined in a.cmake then used in b.cmake. Then we should include a.cmake first. But, the actually link order could be different.

Reply:
Actually, I should not call it ONNXRUNTIME_TARGETS. It is the list of cmake files.